lib/authc/OAuthBasicExchangeAuthenticator.js

```diff
@@ -53,33 +63,72 @@ OAuthBasicExchangeAuthenticator.prototype.authenticate = function authenticate(c
   });
 };
 
-OAuthBasicExchangeAuthenticator.prototype.buildTokenResponse = function buildTokenResponse(apiKey) {
+OAuthBasicExchangeAuthenticator.prototype.buildTokenResponse = function buildTokenResponse(apiKey, callback) {
   var self = this;
-  var scope = self.scopeFactory(apiKey.account,self.requestedScope);
-  // TODO v1.0.0 - remove string option for tokens, should be array only
-  return {
-    "access_token": self.buildAccesstoken(apiKey.account),
-    "token_type":"bearer",
-    "expires_in": self.ttl,
-    "scope": Array.isArray(scope) ? scope.join(' ') : scope
-  };
+
+  var account = apiKey.account;
+  var requestedScope = self.requestedScope;
+
+  function retrieveScopeFactoryResult(callback) {
+    var hasBeenCalled = false;
+
+    function callbackWithResult(err, scope) {
+      if (hasBeenCalled) {
+        throw new Error('Callback has already been called once. Assert that your scopeFactory doesn\'t return a result while also calling the callback.');
+      }
+
+      hasBeenCalled = true;
+
+      if (err) {
+        callback(err);
+        return;
+      }
+
+      callback(null, scope);
```

Comment:
Couldn't you just do `callback(err, scope);` here and skip the `if` statement at line 82?